packages/contracts-bedrock/contracts/dispute/FaultDisputeGame.sol

```diff
@@ -92,12 +105,73 @@ contract FaultDisputeGame is IFaultDisputeGame, Clone {
      * @inheritdoc IFaultDisputeGame
      */
     function step(
-        uint256 _prestateIndex,
+        uint256 _stateIndex,
         uint256 _parentIndex,
-        bytes calldata _stateData,
-        bytes calldata _proof
+        bool _isAttack,
+        bytes calldata,
+        bytes calldata
     ) external {
-        // TODO - Call the VM to perform the execution step.
+        // Steps cannot be made unless the game is currently in progress.
+        if (status != GameStatus.IN_PROGRESS) {
+            revert GameNotInProgress();
+        }
+
+        // Get the parent. If it does not exist, the call will revert with OOB.
+        ClaimData storage parent = claimData[_parentIndex];
+
+        // Pull the parent position out of storage.
+        Position parentPos = parent.position;
+        // Determine the position of the step.
+        Position stepPos = _isAttack ? parentPos.attack() : parentPos.defend();
+
+        // Ensure that the step position is 1 deeper than the maximum game depth.
+        if (stepPos.depth() != MAX_GAME_DEPTH + 1) {
+            revert InvalidParent();
+        }
+
+        // Determine the expected pre & post states of the step.
+        Claim preStateClaim;
+        Claim postStateClaim;
+        if (stepPos.indexAtDepth() == 0) {
+            // If the step position's index at depth is 0, the prestate is the absolute prestate
+            // and the post state is the parent claim.
+            preStateClaim = ABSOLUTE_PRESTATE;
```

Comment:
It makes sense that the `_stateIndex` ends up unused. But maybe we could also assert that input `_stateIndex` equals `ABSOLUTE_PRESTATE`, for extra safety?

Reply:
Is there a need to do this other than for sanity purposes? If the step position's index at depth is zero, there's no other option for the prestate - doing this assertion just adds to the tx cost.

Reply:
It's just for sanity checking. I'm totally also OK with not doing this given the added gas cost.